lib/ts/controllers/s-popover.ts

```diff
@@ -40,6 +40,11 @@ namespace Stacks {
                 // just call initialize here, not show. This keeps already visible popovers from adding/firing document events
                 this.initializePopper();
             }
+
+            if (this.data.get("auto-show") === "true") {
+                this.data.delete("auto-show");
+                this.show();
```

Comment:
Just tested locally and there's a slight issue. I'd expect the `auto-show` to work like our current `is-visible` popovers so we can deprecate that method (this is *far* superior).

The issue I found is that with our current default visible popovers we don't close them on outside clicks until you've interacted with them at least once. We do this by simply not attaching the event handlers at connect and letting them stay shown. Your current implementation causes any auto-shown popovers to hide themselves on document click.

I'd recommend *not* removing the `auto-show` attribute here, check for it in `show`, use it to not trigger `show` event, and then bail after the popper has been initialized, but before `shown()` is called, removing the attribute before bail.

Reply:
This would mean `showPopover` has different behavior if it's called before or after the controller is wired up, which I'd be strongly against.

I think the real problem is that `is-visible` is being used to do something novel and short-circuiting the show logic (I thought it was a bug at first) to keep itself around.  What we really need is something like `data-s-popover-dismiss-on-outside-click="false"`

Reply:
The issue here is that we *want* the popover to dismiss on outside click, just not until the user interacts with it the first time. If the user chooses to dismiss it, then later shows it, we'd want it to act "normal".

I can see a compromise here though where setting `auto-show` or `is-visible` also sets `dismiss=false` automatically (which is where the real regression is here) and just deal with the event not being bound at all and making the user manually dismiss if the popover is reshown (unlikely).

@aaronshekey Thoughts?

Reply:
@b-kelly Maybe I'm misunderstanding.  The current behavior of `is-visible` at https://stackoverflow.design/product/components/popovers/ is that the "Create team" popover can only be dismissed by clicking the dismiss button.  This is achieved by not binding the document level events.

If that's the requirement, adding a setting `data-s-popover-dismiss-on-outside-click` makes things a lot clearer than having a `auto-show` or `is-visible` overloaded with a second unexpected behavior, especially since we are using `auto-show` inside `showPopover` and we want it to behave exactly like it would if the controller instance already existed.